perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/AppAutoRejectionScheduler.java

```diff
@@ -174,43 +197,154 @@ private void rejectExpiredApplications (List<Application> applications, Attribut
+		return getRejectMessage(attribute, lang, key, defaultMsg)
+				.replaceAll(GROUP_PLACEHOLDER, group.getName());
+	}
+
+	/**
+	 * Returns reject message obtained from the given attribute, using the given key; or returns the default msg,
+	 * if there is no suitable message in the attribute. If there is a %vo_name% placeholder, it will
+	 * be replaced by the given vo's name.
+	 *
+	 * @param attribute attribute of type map
+	 * @param lang user preferred language
+	 * @param key key used to find the value in the given attribute
+	 * @param defaultMsg returned if there is no suitable message in the attribute
+	 * @param vo this vo's name will be placed in the message, if there is a %vo_name% placeholder
+	 * @return reject message, concerning the preferred language, obtained from the given attribute,
+	 *         or the given default message
+	 */
+	private String getVoMessage(Attribute attribute, String key, String lang, String defaultMsg, Vo vo) {
+		return getRejectMessage(attribute, lang, key, defaultMsg)
+				.replaceAll(VO_PLACEHOLDER, vo.getName());
+	}
+
+	/**
+	 * Returns reject message obtained from the given attribute, using the given key; or returns the default msg,
+	 * if there is no suitable message in the attribute.
+	 *
+	 * @param attribute attribute of type map
+	 * @param lang user preferred language
+	 * @param key key used to find the value in the given attribute
+	 * @param defaultMsg returned if there is no suitable message in the attribute
+	 * @return reject message, concerning the preferred language, obtained from the given attribute,
+	 *         or the given default message
+	 */
+	private String getRejectMessage(Attribute attribute, String lang, String key, String defaultMsg) {
+		String attrKey = key + "-" + lang;
+		String attrValue = attribute != null  && attribute.getValue() != null
+				? attribute.valueAsMap().get(attrKey)
+				: null;
+
+		// if the version for preferred language is not specified, try to get for en version
+		if (attrValue == null && attribute != null  && attribute.getValue() != null) {
+			attrValue = attribute.valueAsMap().get(key);
+		}
+
+		return attrValue != null ? attrValue : defaultMsg;
+	}
+
+	/**
+	 * Returns preferred language of user from the given application.
+	 * If the user has no preferred language or is null, 'en' is returned.
+	 *
+	 * @param application user
+	 * @return preferred language, 'en' as default
+	 */
+	private String getUserPreferredLang(Application application) {
+		User user = application.getUser();
+		if (user == null) {
+			return "en";
+		}
+		try {
+			Attribute langAttr = perun.getAttributesManagerBl().getAttribute(sess, user, A_USER_PREFERRED_LANGUAGE);
+			String attrValue = langAttr.valueAsString();
+
+			if (attrValue != null && !attrValue.isBlank()) {
+				return attrValue;
+			}
+		} catch (WrongAttributeAssignmentException | AttributeNotExistsException e) {
+			log.error("Failed to read user preferred lang attribute.", e);
 		}
 
+		return "en";
```

Comment:
You should use "defaultLanguage" variable instead of this string (it exists more than once).